ts41/delimiter-cased-props-deep.d.ts

```diff
@@ -0,0 +1,43 @@
+import {DelimiterCase} from './delimiter-case';
+
+/**
+Convert object props to delimiter-case recursively.
+
+This can be useful when, for example, converting some API types from other style.
+
+@see DelimiterCase
+@see DelimiterCasedProps
+@example
+```
+interface User {
+    userId: number;
+    userName: string;
+}
+
+interface UserWithFriends {
+    userInfo: User;
+    userFriends: User[];
+}
+
+const result: DelimiterCasedPropsDeep<UserWithFriends, '-'> = {
+    'user-info': {
+        'user-id': 1,
+        'user-name': "Tom",
+    },
+    'user-friends': [
+        {
+            'user-id': 2,
+            'user-name': "Jerry",
+        },
+        {
+            'user-id': 3,
+            'user-name': "Spike",
+        },
+    ],
+};
+
+```
+*/
+export type DelimiterCasedPropsDeep<T, D extends string> = T extends Array<infer U> ? Array<DelimiterCasedPropsDeep<U, D>> : {
```

Comment:
Use descriptive type names. See existing types for our style.

Reply:
resolved